packages/arb-token-bridge-ui/src/state/app/utils.ts

```diff
@@ -0,0 +1,131 @@
+import dayjs from 'dayjs'
+import { L1ToL2MessageStatus } from '@arbitrum/sdk'
+import { ethers, BigNumber } from 'ethers'
+import {
+  L2ToL1EventResultPlus,
+  Transaction,
+  OutgoingMessageState,
+  getUniqueIdOrHashFromEvent
+} from 'token-bridge-sdk'
+import { DepositStatus, MergedTransaction } from './state'
+
+export const outgoungStateToString = {
+  [OutgoingMessageState.UNCONFIRMED]: 'Unconfirmed',
+  [OutgoingMessageState.CONFIRMED]: 'Confirmed',
+  [OutgoingMessageState.EXECUTED]: 'Executed'
+}
+
+export const getDepositStatus = (tx: Transaction) => {
+  if (tx.type !== 'deposit' && tx.type !== 'deposit-l1') return undefined
+  if (tx.status === 'failure') {
+    return DepositStatus.L1_FAILURE
+  }
+  if (tx.status === 'pending') {
+    return DepositStatus.L1_PENDING
+  }
+  // l1 succeeded...
+  const { l1ToL2MsgData } = tx
+  if (!l1ToL2MsgData) {
+    return DepositStatus.L2_PENDING
+  }
+  switch (l1ToL2MsgData.status) {
+    case L1ToL2MessageStatus.NOT_YET_CREATED:
+      return DepositStatus.L2_PENDING
+    case L1ToL2MessageStatus.CREATION_FAILED:
+      return DepositStatus.CREATION_FAILED
+    case L1ToL2MessageStatus.EXPIRED:
+      return tx.assetType === 'ETH'
+        ? DepositStatus.L2_SUCCESS
+        : DepositStatus.EXPIRED
+    case L1ToL2MessageStatus.FUNDS_DEPOSITED_ON_L2: {
+      return tx.assetType === 'ETH'
+        ? DepositStatus.L2_SUCCESS
+        : DepositStatus.L2_FAILURE
+    }
+    case L1ToL2MessageStatus.REDEEMED:
+      return DepositStatus.L2_SUCCESS
+  }
+}
+
+export const transformDeposits = (
+  deposits: Transaction[]
+): MergedTransaction[] => {
+  return deposits.map(tx => {
+    return {
+      direction: tx.type,
+      status: tx.status,
+      createdAt: tx.timestampCreated
+        ? dayjs(tx.timestampCreated).format('MMM DD, YYYY hh:mm A')
+        : null,
+      createdAtTime: tx.timestampCreated
+        ? dayjs(tx.timestampCreated).toDate().getTime()
```

Comment:
`tx.timestampCreated` should be a timestamp already, why do we parse it with `dayjs`, then to timestamp again?

Reply:
`timestampCreated` is a timestamp but a string. Here we need to display it as a valid human readable date. Hence we use dayjs to transform it.